schema.graphql

```diff
@@ -38,6 +38,48 @@ type Query {
         "Maximum number of recommendations to return in {Slate.recommendations}, defaults to 10"
         recommendationCount: Int = 10
     ): SlateLineup
+
+
+    """
+    Request a specific personalized `SlateLineup` by id. This response allows for fallback behavior when there is not
+    a personalized slate lineup available.
+
+    Examples:
+
+    fragment SlateLineupParts on SlateLineup {
+        id
+        experimentId
+        slates {
+            recommendations {
+                id
+                item: {
+                    resolvedUrl
+                    title
+                    topImageUrl
+                }
+            }
+        }
+    }
+
+    query getPersonalizedSlateLineup($id: String!) {
+        getPersonalizedSlateLineup(id: $id) {
+            personalizedSlateLineup {
+                ...SlateLineupParts
+            }
+            fallbackSlateLineup {
+                ...SlateLineupParts
+            }
+        }
+    }
```

Comment:
I'm into this. I have seen a similar pattern in a number of API responses wherein the JSON always included `data` and `error`, and if there was no `data`, the client would get the `error` field to figure out what went wrong.

This is also clearer than an in-situ replacement without a message, and will result in no more code on the client than checking that the returned lineup's ID matched the requested one would. 

The fact that we send the fallback every time also alleviates a concern about making a second request in the majority of cases (since, as I understand it, unpersonalized behavior will still be our majority case for now).

Finally, the potential drawback I could anticipate with this request (and it's not guaranteed) is payload size. For now I'd actually say ignore that until and unless it becomes an issue. If it _does_, we could discuss options like only populating the fallback if the `personalizedSlateLineup` is not available.

Reply:
@iwsmith @lfishhead @mmiermans thoughts?

Reply:
Personally I don't like this solution. Sending both the personalized and unpersonalized responses seems ripe for confusion from clients and couples clients to the implementation details of what kind of recommendations are being generated. Furthermore, what will the clients do with this information (i.e. that the personalized request failed)? They can't fix this problem, there is no action they or their users can take. In practice they are just going to check for one field, and if it is missing use the other field instead. So I don't see why we wouldn't save them the trouble and potential confusion and do it server side.

As for "when a client requests a specific piece of content, we should return an error if that item doesn't exist", there is a philosophical question about what a SlateLineup is. Originally, I created them with the intent of identifying a specific place in a specific client where content is displayed. In this respect personalization doesn't matter, that is an implementation detail for RecsAPI. Similarly, the order the slates are returned in is an implementation detail. All the clients care about is that suitable content for this location is returned. We as RecsAPI care, but the clients shouldn't. 

Note that there has been a proliferation of SlateLineups, so that many different ones can now represent that same client/location. I think that is b/c we ended up using new SlateLineups to run experiments rather than setting them up as an experiment inside the SlateLineup (I suspect this is b/c we never implemented that functionality in RecsAPI).

Reply:
Ah reading below I see that the intent here is to use a typed union. If so just ignore most of what I wrote above.